File: VB/E558/HourRecurrenceControl.vb
Imports DevExpress.XtraScheduler.UI
Imports System

Namespace E558
    Partial Public Class HourRecurrenceControl
        Inherits RecurrenceControlBase

        Public Sub New()
            MyBase.New()
            InitializeComponent()
            Me.UpdateControlsCore()
            Me.SubscribeControlsEvents()

        End Sub
        Protected Overrides Sub SubscribeControlsEvents()
            MyBase.SubscribeControlsEvents()
            AddHandler spinEdit1.EditValueChanged, AddressOf spinEdit1_EditValueChanged
        End Sub
        Protected Overrides Sub UnsubscribeControlsEvents()
            MyBase.UnsubscribeControlsEvents()
            RemoveHandler spinEdit1.EditValueChanged, AddressOf spinEdit1_EditValueChanged
        End Sub

        Private Sub spinEdit1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            MyBase.RecurrenceInfo.Periodicity = MyBase.Validator.GetIntegerValue(Me.spinEdit1.EditValue)
        End Sub
        Protected Overrides Sub UpdateControlsCore()
            MyBase.UpdateControlsCore()
            Me.spinEdit1.EditValue = MyBase.RecurrenceInfo.Periodicity
        End Sub
    End Class
End Namespace

File: CS/E558/HourRecurrenceControl.cs
using DevExpress.XtraScheduler.UI;
using System;

namespace E558
{
    public partial class HourRecurrenceControl : RecurrenceControlBase
    {
        public HourRecurrenceControl():base()
        {
            InitializeComponent();
            this.UpdateControlsCore();
            this.SubscribeControlsEvents();

        }
        protected override void SubscribeControlsEvents()
        {
            base.SubscribeControlsEvents();
            this.spinEdit1.EditValueChanged += new EventHandler(spinEdit1_EditValueChanged);
        }
        protected override void UnsubscribeControlsEvents()
        {
            base.UnsubscribeControlsEvents();
            this.spinEdit1.EditValueChanged -= new EventHandler(spinEdit1_EditValueChanged);
        }

        void spinEdit1_EditValueChanged(object sender, EventArgs e)
        {
            base.RecurrenceInfo.Periodicity = base.Validator.GetIntegerValue(this.spinEdit1.EditValue);
        }
        protected override void UpdateControlsCore()
        {
            base.UpdateControlsCore();
            this.spinEdit1.EditValue = base.RecurrenceInfo.Periodicity;
        }
    }
}
